editor/draw.js

```diff
@@ -31,31 +33,150 @@ var RandomizeModes = {
 var randomize_ids = RandomizeModes.LET_DOCUMENT_DECIDE;
 
 /**
- * This class encapsulates the concept of a layer in the drawing
- * @param {String} name - Layer name
- * @param {SVGGElement} child - Layer SVG group.
+ * Add class 'layer' to the element
+ *
+ * Parameters:
+ * @param {SVGGElement} elem - The SVG element to update
+ */
+function addLayerClass(elem) {
+	var classes = elem.getAttribute('class');
+	if (classes === null || classes === undefined || classes.length === 0) {
+		elem.setAttribute('class', LAYER_CLASS);
+	} else if (! LAYER_CLASS_REGEX.test(classes)) {
+		elem.setAttribute('class', classes + ' ' + LAYER_CLASS);
+	}
+}
+
+function createLayer(name, svgElem) {
+	if (!svgElem) {
+		return undefined;
+	}
+	var svgdoc = svgElem.ownerDocument;
+	var new_layer = svgdoc.createElementNS(NS.SVG, "g");
+	var layer_title = svgdoc.createElementNS(NS.SVG, "title");
+	layer_title.textContent = name;
+	new_layer.appendChild(layer_title);
+	svgElem.appendChild(new_layer);
+	return new_layer;
+}
```

Comment:
This is a little confusing.  We have a createLayer() function, then we have a Layer constructor which calls it, then there is a Drawing.createLayer().  It seems like a Layer should know how to create itself, can we make it part of the constructor?

If that's too much to refactor again, just say so - I like this patch overall and am ready to merge it.